src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/AdditionalConditionWithConfig.kt

```diff
@@ -0,0 +1,33 @@
+package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model
+
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.entity.AdditionalCondition
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.policy.AdditionalConditions
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.policy.IAdditionalCondition
+
+data class AdditionalConditionWithConfig(
+  val additionalCondition: AdditionalCondition,
+  val config: IAdditionalCondition,
+)
+
+fun mapConditionsToConfig(licenceConditions: List<AdditionalCondition>, policy: AdditionalConditions): List<AdditionalConditionWithConfig> {
+  val policyAdditionalConditions = policy.ap + policy.pss
+  return licenceConditions.map {
+    val policyCondition = policyAdditionalConditions.find { pc -> it.conditionCode == pc.code }!!
+    AdditionalConditionWithConfig(it, policyCondition)
+  }
+}
+
+fun checkConditionsReadyToSubmit(licenceConditions: List<AdditionalCondition>, policy: AdditionalConditions): Map<String, Boolean> {
+  if (licenceConditions.isEmpty()) { return emptyMap() }
```

Comment:
don't need this guard, an empty list is handled naturally anyway

Reply:
Ah, I assume the not-null check in `mapConditionsToConfig` would complain if the list was empty, as I thought `find` would return null on an empty list
`val policyCondition = policyAdditionalConditions.find { pc -> it.conditionCode == pc.code }!!`